prometheuscollector/prometheuscollector.go

```diff
@@ -46,11 +46,11 @@ var (
 )
 
 type Collector struct {
-	metrics tusd.Metrics
+	metrics *tusd.Metrics
```

Comment:
Good catch! However, I would rather address this issue at the root, meaning `newErrorsTotalMap` should return a pointer. This should have the same effect and would prevent the lock from being copied.